markslack/__init__.py

```diff
@@ -7,24 +7,37 @@
     '|[!*\(\),]|(?:%[0-9a-fA-F][0-9a-fA-F]))+'
 )
 
+emoji_pattern = u':[a-zA-Z0-9\+\-_&.ô’Åéãíç()!#*]+:'
+
 
 class MarkSlack(object):
     def __init__(
         self,
         markslack_links=True,
+        replace_emojis=True,
         link_templates=None,
         user_templates=None,
         image_template=None,
         image_extensions=['.jpg', '.png'],
     ):
         self.user_templates = user_templates
         self.markslack_links = markslack_links
+        self.replace_emojis = replace_emojis
         self.image_extensions = image_extensions
         self.image_template = image_template
         self.link_templates = link_templates
 
     def mark_emoji(self):
-        self.marked = emoji.emojize(self.marked, use_aliases=True)
+        if self.replace_emojis:
+            self.marked = emoji.emojize(self.marked, use_aliases=True)
+
+        # Emojis at the end of text
+        p1 = re.compile(u'(\s*)((?:{0})+)(\s*|\.*)$'.format(emoji_pattern))
```

Comment:
I like this. I wonder, though, if we should make replacement another config option on the MarkSlack class. For example, sometimes we want to pass through emoji handles to a downstream renderer like [this](https://github.com/tommoor/react-emoji-render).

So in that case, we'd want to preserve `:fake_emoji:` in the test and handle it later.

Maybe the option could be `remove_emoji=False`?

Whatchu think?

Reply:
Hmm I guess I can see that. But we probabaly don't want to call the option `remove_emoji`. It would be something like `remove_unsupported_emoji`. Because we would want it to only remove the emoji that weren't already replaced with their real icon.

Reply:
`remove_bad_emoji`?